api/capi/src/nntrainer.cpp

```diff
@@ -138,14 +138,19 @@ static int ml_train_dataset_create(ml_train_dataset_h *dataset,
   int status = ML_ERROR_NONE;
 
   check_feature_state();
+  if (dataset == NULL) {
+    return ML_ERROR_INVALID_PARAMETER;
+  }
 
   ml_train_dataset *nndataset = new ml_train_dataset;
   nndataset->magic = ML_NNTRAINER_MAGIC;
   nndataset->in_use = false;
 
   returnable f = [&]() {
-    nndataset->dataset[ML_TRAIN_DATASET_DATA_USAGE_TRAIN] =
-      ml::train::createDataset(type, train);
+    if (train != nullptr) {
```

Comment:
train is a must as far as I know.

Reply:
Yes, before delegating this function it is checked. This is to share the logic between making an empty dataset and the old way.

Reply:
aah I missed that.